transport/src/main/java/io/netty/channel/DefaultChannelConfig.java

```diff
@@ -227,6 +258,85 @@ public ChannelConfig setMaxMessagesPerRead(int maxMessagesPerRead) {
     }
 
     @Override
+    public int getMaxBytesPerRead() {
+        return maxBytesPerRead;
+    }
+
+    @Override
+    public ChannelConfig setMaxBytesPerRead(int maxBytesPerRead) {
+        if (maxBytesPerRead <= 0) {
+            throw new IllegalArgumentException("maxBytesPerRead: " + maxBytesPerRead + " (expected: > 0)");
+        }
+        // There is a dependency between this.maxBytesPerRead and this.maxBytesPerIndividualRead (a < b).
+        // Write operations must be synchronized, but independent read operations can just be volatile.
+        synchronized (this) {
+            final int maxBytesPerIndividualRead = getMaxBytesPerIndividualRead();
+            if (maxBytesPerRead < maxBytesPerIndividualRead) {
+                throw new IllegalArgumentException(
+                        "maxBytesPerRead cannot be less than " +
+                                "maxBytesPerIndividualRead (" + maxBytesPerIndividualRead + "): " + maxBytesPerRead);
+            }
+
+            this.maxBytesPerRead = maxBytesPerRead;
+        }
+        return this;
+    }
+
+    @Override
+    public int getMaxBytesPerIndividualRead() {
+        return maxBytesPerIndividualRead;
+    }
+
+    @Override
+    public ChannelConfig setMaxBytesPerIndividualRead(int maxBytesPerIndividualRead) {
+        if (maxBytesPerIndividualRead <= 0) {
+            throw new IllegalArgumentException(
+                    "maxBytesPerIndividualRead: " + maxBytesPerIndividualRead + " (expected: > 0)");
+        }
+        // There is a dependency between this.maxBytesPerRead and this.maxBytesPerIndividualRead (a < b).
+        // Write operations must be synchronized, but independent read operations can just be volatile.
+        synchronized (this) {
+            final int maxBytesPerRead = getMaxBytesPerRead();
+            if (maxBytesPerIndividualRead > maxBytesPerRead) {
+                throw new IllegalArgumentException(
+                        "maxBytesPerIndividualRead cannot be greater than " +
+                                "maxBytesPerRead (" + maxBytesPerRead + "): " + maxBytesPerIndividualRead);
+            }
+
+            this.maxBytesPerIndividualRead = maxBytesPerIndividualRead;
+        }
+        return this;
+    }
+
+    @Override
+    public synchronized Entry<Integer, Integer> getMaxBytesPerReadPair() {
+        return new AbstractMap.SimpleEntry<Integer, Integer>(maxBytesPerRead, maxBytesPerIndividualRead);
+    }
+
+    @Override
+    public ChannelConfig setMaxBytesPerReadPair(int maxBytesPerRead, int maxBytesPerIndividualRead) {
```

Comment:
I think something similar can be done to resolve https://github.com/netty/netty/issues/3806.